libarchive/test/test_write_format_zip_entry_size_unset.c

```diff
@@ -107,10 +107,22 @@ static void write_archive(struct archive *a)
 
 static void verify_contents(const char *zip_buff, size_t size)
 {
-	unsigned long crc = bitcrc32(0, file_data1, sizeof(file_data1));
-	crc = bitcrc32(crc, file_data2, sizeof(file_data2));
+	const unsigned long crc = bitcrc32(
+			bitcrc32(0, file_data1, sizeof(file_data1)), file_data2, sizeof(file_data2)
+	);
 
 	const char *zip_end = zip_buff + size;
+	/* Fix incompatible-pointer-types-discards-qualifiers */
```

Comment:
From quick look, I cannot see where/why that is the case. If needed please consider fixing the API to use `const char *` or if that's not possible use local casting.

The union seems like a pretty gross hack.